plugins/pipelines/app/models/concerns/samson_pipelines/stage_concern.rb

```diff
@@ -8,6 +10,34 @@ def next_stages
     Stage.find(next_stage_ids)
   end
 
+  # Needs to find all the possible stages in case this is a pipeline of pipelines as each subsequent stage
+  # could have valid next_stage_ids
+  def all_stages
+    stage_collection = []
+    stages = Stage.find(next_stage_ids)
+    stages.each do |this_stage|
+      stage_collection.push(this_stage.id)
+      stage_collection += recursive_stages(this_stage.id)
+    end
+    stage_collection.flatten!
+    stage_collection.map { |id| id.to_i }
+    Stage.find(stage_collection);
+  end
+
+  def recursive_stages(parent_stage_id)
+    result = []
+    stages_ids = Stage.where(id: parent_stage_id).map(&:next_stage_ids)
+    return [parent_stage_id] if stages_ids.empty?
+    stages_ids.each do |stage_id|
+        result << recursive_stages(stage_id)
+    end
+    result << parent_stage_id
+  end
+
+  def deploy_requires_approval?
```

Comment:
--> #0  SamsonPipelines::StageConcern.deploy_requires_approval? at /Users/iwaters/co/zendesk/samson/plugins/pipelines/app/models/concerns/samson_pipelines/stage_concern.rb:34
    #1  Deploy.waiting_for_buddy? at /Users/iwaters/co/zendesk/samson/app/models/deploy.rb:83
    #2  DeployService.deploy!(stage#Stage, attributes#Hash) at /Users/iwaters/co/zendesk/samson/app/models/deploy_service.rb:15

is how it get's called seemingly.